src/msg.c

```diff
@@ -493,12 +495,11 @@ static int8_t msg_register(struct session *session,
 			   const knot_msg_register *kreq, size_t ilen,
 			   knot_msg_credential *krsp)
 {
-	struct knot_device *device;
+	struct mydevice *mydevice = l_new(struct mydevice, 1);
```

Comment:
What do you think about change mydevice to device_pending?

Reply:
Done